Fixed optimized inventory system

sakipvp opened the pull request.

SDClowen commented:
You don't want to understand me. Your code causes various bugs. I was already working on some bugs from the previous pull-request. These files cause conflicts and your edits do not lead to any resolution. You said optimized, but it's nothing but using more resources. If you are looking for a task to work on, you can work on Quests. For example like in mbot. I look forward to the continuation of your admirable work.

sakipvp commented:
Thank you for these words: admirable work.
But I think differently about this pull-request:
1. The most important, after lots of tests and parse packets(inventory operations): I fixed bugs in InventoryItemCollection Move and MoveTo methods(and today I found a new one)
2. I optimized Storage(removed unnecessary things) and AbilityPet(in FromPacket method used InventoryItemCollection(packet) constructor instead of ParseInventory method, and changed ParseInventory method to UpdateInventorySize and used it in UpdateResponse at OnUpdateAbilityPetInventorySize).
3. I fixed namespaces in the RSBot.Core.Objects.Inventory.
I understand that if you don't agree with me(if possible with some words of justification), but why don't you fix bugs based on 1 point.
So please tell me, how my code is causing the errors.

SDClowen commented:
First of all, I am already working on the inventory system. Your and my files are conflicting and once I'm done you'll see what your errors are.
